pkg/asset/agent/manifests/agent_config_type.go

```diff
@@ -0,0 +1,33 @@
+package manifests
+
+import (
+	metal3v1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AgentConfig is the API for specifying additional configuration for the
+// agent-based installer not covered by install-config.
+type AgentConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AgentConfigSpec `json:"spec,omitempty"`
+	// No status
+}
+
+// AgentConfigSpec contains additional configuration for the agent-based installer
+type AgentConfigSpec struct {
+	// node0 hostname
+	Node0 string            `json:"node0,omitempty"`
```

Comment:
(bikeshed alert) I have questions over whether `node0` is going to be the most meaningful name for users, even though it's been helpful for us.
I suggested `rendezvousIP` but I'm open to something easier to spell.

@celebdor are we confident that we can always accept a hostname here? I have a suspicion that there are circumstances where we will require an IP (that's certainly all we support at the moment), and perhaps we should lean in to this.

Reply:
I'm not tied to it being node0. It was defined as node0 in the example agent-config from a month ago.